### ISSUE TITLE
Question: How to globally inject components with Vue3?

### ISSUE THREAD

**Issue by ccmetz** [vuejs/vue-test-utils-next#1043]
I'm using `@vue/test-utils` (v2.0.0-0) with a Vue3 + TS project that uses `Vuestic-UI` as a components framework. I'm trying to figure out how to inject my `Vuestic-UI` components globally so that I don't get messages like the following when I try to run a unit test (using `jest` as the test framework btw):
```
console.warn node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6799
    [Vue warn]: Failed to resolve component: va-button 
      at <Anonymous ref="VTU_COMPONENT" > 
      at <VTUROOT>
```
In the past with Vue2, I was able to use a setup file with something like this to import all of the components of the framework I was using (at the time it was `Buefy`):
```
import Vue from 'vue'
import Buefy from 'buefy'

// We need to register Buefy so that jest recognizes their components during our tests
Vue.use(Buefy);
```

How do I go about doing this with Vue3? I haven't been able to find any documentation about it though there is this page here that looks like it might've helped: https://next.vue-test-utils.vuejs.org/guide/advanced/third-party.html. Any help is much appreciated :)

**Comment by vpillinger** [comment]
I ran into a similar issue here and the challenge for me was that the globals are registered using `app.component()` in Vue, but using this array syntax in the testing library. Luckily, I was using a custom-built component library so I was able to create an "index" that I could import to this library. I then used a for-loop to register the components in my `main.js` file.

It seems like the confusion/issue here is that this library handles global registration in a completely different way from Vue itself.

**Comment by lmiller1990** [comment]
It seems like PrimeVue asks users to import and register the components one by one: 
![image](https://user-images.githubusercontent.com/19196536/141650331-0131f340-1b8f-42f6-b935-bc9e9d3012cb.png)


This seems very weird. I wonder why they don't register them all globally like other UI libs? Seems this behavior is correct and intentional - at least, I don't see a bug in Test Utils (assuming this is really how PrimeVue works?)

**Comment by lmiller1990** [comment]
I got a Prime Vue example here: https://github.com/vuejs/vue-test-utils-next/compare/add-prime-vue-example?expand=1

Had the same problem as @awulkan. For whatever reason PrimeVue ship `vue` files, and Jest won't transform `vue` files in node_modules by default. In my example above, I just imported the `.cjs.js` file and it worked. You could also use the `vue` files by configure [this option](https://jestjs.io/docs/configuration#transformignorepatterns-arraystring) in Jest and telling to to apply the Vue transform to any Vue files in `node_modules/primevue`.

In short, I don't think there's a bug; PrimeVue is odd in that it doesn't globally register the components when you install it. I suspect the same problem is happening with Vuestic-UI. We could make a PR to those projects support an option to install all components globally, perhaps? Unless I'm missing something completely. How are you installing PrimeVue and Vuestic-UI in your projects?

**Comment by gibranamparan** [comment]
Hi, I am using Vue3 + Vuestic + Vitest
This is how injected vuestic components to my unit test.
Under vite.config.ts I have the config definition with a test property with an alias in definint vuestic-ui this way
```javascript
defineConfig({
    plugins: [
      vue(),
    ],
    test: {
      environment: "jsdom",
      alias: {
        'vuestic-ui': 'vuestic-ui/dist/esm/main.js'
      },
    },`

Then in my test.ts file, for example for a test button I create a vuestic instance and use mount options property to mount my custom component with the mount method of vue-test-utils.
```javascript
import {  createVuestic } from 'vuestic-ui'
import vuesticGlobalConfig from '../src/services/vuestic-ui/global-config'
const vuestic = createVuestic({ config: vuesticGlobalConfig })

const initComponent = () => {
  return mount(TestButton, {
    global: {
      plugins: [vuestic]
    },
  })
}`
Hopefully this works for you.
